### PR TITLE
Add noDuplicateKeys lint rule

### PR DESCRIPTION
**Pull request by mattsjones**

### CODE REVIEW

File: packages/@romejs/js-compiler/transforms/lint/noDuplicateKeys.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ */
+
+import {Path} from '@romejs/js-compiler';
+import {ObjectProperty, ObjectMethod, SpreadProperty} from '@romejs/js-ast';
+import {TransformExitResult} from '@romejs/js-compiler/types';
+import {markup} from '@romejs/string-markup';
+
+function extractPropertyKey(
+  node: ObjectProperty | ObjectMethod | SpreadProperty,
+) {
```
Comment (sebmck) [inline packages/@romejs/js-compiler/transforms/lint/noDuplicateKeys.ts]:
```suggestion
): undefined | string {
```

File: packages/@romejs/js-compiler/transforms/lint/noDuplicateKeys.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    if (value.type === 'PrivateName') {
+      return value.id.name;
+    }
+
+    if (value.type === 'Identifier') {
+      return value.name;
+    }
+
+    return value.value;
```
Comment (sebmck) [inline packages/@romejs/js-compiler/transforms/lint/noDuplicateKeys.ts]:
```suggestion
    return String(value.value);
```

File: packages/@romejs/js-compiler/transforms/lint/noDuplicateKeys.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  return null;
```
Comment (sebmck) [inline packages/@romejs/js-compiler/transforms/lint/noDuplicateKeys.ts]:
```suggestion
  return undefined;
```

File: packages/@romejs/js-compiler/transforms/lint/noDuplicateKeys.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  name: 'noDuplicateKeys',
+  enter(path: Path): TransformExitResult {
+    const {node} = path;
+    const previousKeys = new Set();
+
+    if (node.type === 'ObjectExpression') {
+      for (const prop of node.properties) {
+        const key = extractPropertyKey(prop);
+
+        if (key !== null) {
```
Comment (sebmck) [inline packages/@romejs/js-compiler/transforms/lint/noDuplicateKeys.ts]:
```suggestion
        if (key !== undefined) {
```

**Comment by mattsjones** [comment]
@sebmck Addressed all feedback and updated against master 👍

**Comment by sebmck** [comment]
Perfect, thank you!

**Pull request merged**